app/Http/Controllers/Webapp/PostController.php

```diff
@@ -0,0 +1,1890 @@
+        $post->priority = (int)$request->get('post_priority', 0);
+        $post->word_count = (int)$request->get('word_count', 0);
+
+        $submitType = strpos($request->get('submit_type'), '公開') !== false ? 'publish' : 'draft';
+
+        $post->save();
+
+        $post->saveHistory();
+
+        try {
+            if(!$this->currentUser->is(User::CONTRIBUTOR) && $post->site_id > 0) {
+                $apiUrl = Site::find($post->site_id)->api_url;
+                $returnData = (new CdsApiDriver($apiUrl, [
+                    'content' => $post->content,
+                    'title' => $postTitle,
+                    'categories' => [$post->category],
+                    'post_excerpt' => $postDescription,
+                    'post_status' => $submitType,
+                    'feature_img' => $post->feature_img,
+                    'author_name' => !empty($author = User::find($post->user_id)) ? $author->username : '',
+                    'cback_url' => action('Webapp\\PostController@getEdit', $post->id),
+                    'priority' => $post->priority,
+                ]))->send('publish')->getResult();
+
+                if($returnData->data->post_id == 0) {
+                    throw new \Exception();
+                }
+                $post->content = Post::replaceIDButtonLink($returnData->data->post_id, $post->content);
+                $post->target_post_id = $returnData->data->post_id;
+                $post->published_status = ($submitType == 'publish') ? 1 : 0;
+            }
+
+            $this->_flashMessage($request, $submitType == 'publish' ? '公開しました' : '下書きに保存しました', 'success');
+        } catch(\Exception $e) {
+            $post->published_status = 0;
+            $this->_flashMessage($request, '更新に失敗しました。もう一度ご確認ください。', 'warning');
+        }
+
+        $post->save();
+
+        // Save kw and match post_id and kw_id
+        // Main keyword
+        $main = explode(',', $request->get('mainkw'));
+        $uniqueMainKeyword = trim($main[0]);
+        $mainKeyword = Keywords::getKeyword($uniqueMainKeyword, 1);
+
+        if(empty($mainKeyword)) {//there is no keyword
+            if(strlen($uniqueMainKeyword) > 0) {
+                // Insert new main keyword
+                $mainKeyword = new Keywords();
+                $mainKeyword->keyword = $uniqueMainKeyword;
+                $mainKeyword->type = 1;
+                $mainKeyword->save();
+            }
+        }
+
+        $postMetaMkw = PostMeta::getPostMeta($post->id, isset($mainKeyword->id) ? $mainKeyword->id : 0);
+        if(empty($postMetaMkw) && isset($mainKeyword->id)) {
+            // Insert new post-keyword relationship if not exist
+            $postMetaMkw = new PostMeta();
+            $postMetaMkw->post_id = $post->id;
+            $postMetaMkw->keyword_id = $mainKeyword->id;
+            $postMetaMkw->save();
+        }
+
+        // Sub keywords
+        $subs = explode(',', $request->get('subkws'));
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "subkws" 6 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar.relipa.blog/coding_rules#rule_key=php%3AS1192)